### ISSUE TITLE
'charmap' codec error

### ISSUE THREAD

**Issue by guanqun** [wbond/package_control#1273]
I'm using SublimeText 3 and when trying to install the package control, it seems OK, but when I hit "Install Package", the console complains and nothing shows up. The last line of error is something about "charmap codec error", full log as follows:

```
>>> import urllib.request,os,hashlib; h = '6f4c264a24d933ce70df5dedcf1dcaee' + 'ebe013ee18cced0ef93d5f746d80ef60'; pf = 'Package Control.sublime-package'; ipp = sublime.installed_packages_path(); urllib.request.install_opener( urllib.request.build_opener( urllib.request.ProxyHandler()) ); by = urllib.request.urlopen( 'http://packagecontrol.io/' + pf.replace(' ', '%20')).read(); dh = hashlib.sha256(by).hexdigest(); print('Error validating download (got %s instead of %s), please try manual install' % (dh, h)) if dh != h else open(os.path.join( ipp, pf), 'wb' ).write(by)
286331
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading settings Packages/Package Control/Package Control.sublime-settings
reloading plugin 0_package_control_loader.00-package_control
Package Control: Settings change detected, clearing cache
Package Control: Removed directory for unneeded dependency golangconfig
Package Control: Removed directory for unneeded dependency pygments
Package Control: Removed directory for unneeded dependency pyyaml
Package Control: Removed directory for unneeded dependency package_events
Package Control: Removed directory for unneeded dependency shellenv
Package Control: Removed directory for unneeded dependency newterm
Package Control: Installing 8 missing packages
Package Control: Exported certificate: Common Name: thawte Primary Root CA - G3; Organizational Unit: (c) 2008 thawte, Inc. - For authorized use only, Certification Services Division; Organization: thawte, Inc.; Country: US
Package Control: Exported certificate: Common Name: DST ACES CA X6, Organizational Unit: DST ACES, Organization: Digital Signature Trust, Country: US
Package Control: Exported certificate: Common Name: Staat der Nederlanden Root CA - G2, Organization: Staat der Nederlanden, Country: NL
Package Control: Exported certificate: Common Name: Certigna, Organization: Dhimyotis, Country: FR
Package Control: Exported certificate: Common Name: Hongkong Post Root CA 1, Organization: Hongkong Post, Country: HK
Package Control: Exported certificate: Common Name: GeoTrust Primary Certification Authority, Organization: GeoTrust Inc., Country: US
Package Control: Exported certificate: Common Name: Common Policy, Organizational Unit: FBCA, Organization: U.S. Government, Country: us
Package Control: Exported certificate: Common Name: ComSign Global Root CA, Organization: ComSign Ltd., Country: IL
Package Control: Exported certificate: Common Name: thawte Primary Root CA; Organizational Unit: (c) 2006 thawte, Inc. - For authorized use only, Certification Services Division; Organization: thawte, Inc.; Country: US
Package Control: Exported certificate: Organizational Unit: Security Communication RootCA2; Organization: SECOM Trust Systems CO.,LTD.; Country: JP
Package Control: Exported certificate: Common Name: TÜBİTAK UEKAE Kök Sertifika Hizmet Sağlayıcısı - Sürüm 3; Organizational Unit: Kamu Sertifikasyon Merkezi, Ulusal Elektronik ve Kriptoloji Araştırma Enstitüsü - UEKAE; Organization: Türkiye Bilimsel ve Teknolojik Araştırma Kurumu - TÜBİTAK; Locality: Gebze - Kocaeli; Country: TR
Package Control: Exported certificate: Common Name: Izenpe.com, Organization: IZENPE S.A., Country: ES
Package Control: Exported certificate: Organizational Unit: Trustis FPS Root CA, Organization: Trustis Limited, Country: GB
Package Control: Exported certificate: Common Name: SwissSign Platinum CA - G2, Organization: SwissSign AG, Country: CH
Package Control: Exported certificate: Common Name: SwissSign Gold Root CA - G3, Organization: SwissSign AG, Country: CH
Package Control: Exported certificate: Common Name: QuoVadis Root CA 3, Organization: QuoVadis Limited, Country: BM
Package Control: Exported certificate: Common Name: Go Daddy Root Certificate Authority - G2; Organization: GoDaddy.com, Inc.; Locality: Scottsdale; State/Province: Arizona; Country: US
Package Control: Exported certificate: Common Name: Sonera Class1 CA, Organization: Sonera, Country: FI
Package Control: Exported certificate: Common Name: TWCA Global Root CA, Organizational Unit: Root CA, Organization: TAIWAN-CA, Country: TW
Package Control: Exported certificate: Common Name: Certinomis - Root CA, Organizational Unit: 0002 433998903, Organization: Certinomis, Country: FR
Package Control: Exported certificate: Email Address: Info@izenpe.com, Common Name: Izenpe.com, Locality: Avda del Mediterraneo Etorbidea 3 - 01010 Vitoria-Gasteiz, Organization: IZENPE S.A. - CIF A-01337260-RMerc.Vitoria-Gasteiz T1055 F62 S8, Country: ES
Package Control: Exported certificate: Organizational Unit: ePKI Root Certification Authority; Organization: Chunghwa Telecom Co., Ltd.; Country: TW
Package Control: Exported certificate: Common Name: T-TeleSec GlobalRoot Class 2, Organizational Unit: T-Systems Trust Center, Organization: T-Systems Enterprise Services GmbH, Country: DE
Package Control: Exported certificate: Common Name: GeoTrust Primary Certification Authority - G3, Organizational Unit: (c) 2008 GeoTrust Inc. - For authorized use only, Organization: GeoTrust Inc., Country: US
Package Control: Exported certificate: Common Name: E-Tugra Certification Authority, Organizational Unit: E-Tugra Sertifikasyon Merkezi, Organization: E-Tuğra EBG Bilişim Teknolojileri ve Hizmetleri A.Ş., Locality: Ankara, Country: TR
Package Control: Exported certificate: Common Name: Global Chambersign Root, Organizational Unit: http://www.chambersign.org, Organization: AC Camerfirma SA CIF A82743287, Country: EU
Package Control: Exported certificate: Common Name: Chambers of Commerce Root - 2008, Organization: AC Camerfirma S.A., Serial Number: A82743287, Locality: Madrid (see current address at www.camerfirma.com/address), Country: EU
Package Control: Exported certificate: Common Name: UTN-USERFirst-Hardware, Organizational Unit: http://www.usertrust.com, Organization: The USERTRUST Network, Locality: Salt Lake City, State/Province: UT, Country: US
Package Control: Exported certificate: Common Name: COMODO RSA Certification Authority, Organization: COMODO CA Limited, Locality: Salford, State/Province: Greater Manchester, Country: GB
Package Control: Exported certificate: Common Name: Autoridad de Certificacion Firmaprofesional CIF A62634068, Country: ES
Package Control: Exported certificate: Email Address: acraiz@suscerte.gob.ve, Organizational Unit: Superintendencia de Servicios de Certificacion Electronica, Organization: Sistema Nacional de Certificacion Electronica, State/Province: Distrito Capital, Locality: Caracas, Country: VE, Common Name: Autoridad de Certificacion Raiz del Estado Venezolano
Package Control: Exported certificate: Common Name: GlobalSign, Organization: GlobalSign, Organizational Unit: GlobalSign Root CA - R2
Package Control: Exported certificate: Common Name: USERTrust ECC Certification Authority, Organization: The USERTRUST Network, Locality: Jersey City, State/Province: New Jersey, Country: US
Package Control: Exported certificate: Common Name: AffirmTrust Commercial, Organization: AffirmTrust, Country: US
Package Control: Exported certificate: Common Name: CA Disig Root R2, Organization: Disig a.s., Locality: Bratislava, Country: SK
Package Control: Exported certificate: Common Name: OISTE WISeKey Global Root GB CA, Organizational Unit: OISTE Foundation Endorsed, Organization: WISeKey, Country: CH
Package Control: Exported certificate: Common Name: Certum Trusted Network CA, Organizational Unit: Certum Certification Authority, Organization: Unizeto Technologies S.A., Country: PL
Package Control: Exported certificate: Common Name: Symantec Class 3 Public Primary Certification Authority - G4, Organizational Unit: Symantec Trust Network, Organization: Symantec Corporation, Country: US
Package Control: Exported certificate: Common Name: Entrust Root Certification Authority - G2; Organizational Unit: (c) 2009 Entrust, Inc. - for authorized use only, See www.entrust.net/legal-terms; Organization: Entrust, Inc.; Country: US
Package Control: Exported certificate: Common Name: Echoworx Root CA2, Organizational Unit: Certification Services, Organization: Echoworx Corporation, Locality: Toronto, State/Province: Ontario, Country: CA
Package Control: Exported certificate: Common Name: StartCom Certification Authority, Organizational Unit: Secure Digital Certificate Signing, Organization: StartCom Ltd., Country: IL
Package Control: Exported certificate: Common Name: CFCA EV ROOT, Organization: China Financial Certification Authority, Country: CN
Package Control: Exported certificate: Organization: TÜRKTRUST Bilgi İletişim ve Bilişim Güvenliği Hizmetleri A.Ş. (c) Aralık 2007, Locality: Ankara, Country: TR, Common Name: TÜRKTRUST Elektronik Sertifika Hizmet Sağlayıcısı
Package Control: Exported certificate: Common Name: Apple Root CA - G2, Organizational Unit: Apple Certification Authority, Organization: Apple Inc., Country: US
Package Control: Exported certificate: Organizational Unit: certSIGN ROOT CA, Organization: certSIGN, Country: RO
Package Control: Exported certificate: Common Name: D-TRUST Root Class 3 CA 2 2009, Organization: D-Trust GmbH, Country: DE
Package Control: Exported certificate: Common Name: SZAFIR ROOT CA, Organization: Krajowa Izba Rozliczeniowa S.A., Country: PL
Package Control: Exported certificate: Common Name: Symantec Class 1 Public Primary Certification Authority - G6, Organizational Unit: Symantec Trust Network, Organization: Symantec Corporation, Country: US
Package Control: Exported certificate: Common Name: Deutsche Telekom Root CA 2, Organizational Unit: T-TeleSec Trust Center, Organization: Deutsche Telekom AG, Country: DE
Package Control: Exported certificate: Common Name: Starfield Root Certificate Authority - G2; Organization: Starfield Technologies, Inc.; Locality: Scottsdale; State/Province: Arizona; Country: US
Package Control: Exported certificate: Common Name: SwissSign Platinum Root CA - G3, Organization: SwissSign AG, Country: CH
Package Control: Exported certificate: Organizational Unit: Security Communication RootCA1, Organization: SECOM Trust.net, Country: JP
Package Control: Exported certificate: Common Name: Chambers of Commerce Root, Organizational Unit: http://www.chambersign.org, Organization: AC Camerfirma SA CIF A82743287, Country: EU
Package Control: Exported certificate: Common Name: TWCA Root Certification Authority, Organizational Unit: Root CA, Organization: TAIWAN-CA, Country: TW
Package Control: Exported certificate: Common Name: AffirmTrust Premium ECC, Organization: AffirmTrust, Country: US
Package Control: Exported certificate: Email Address: pki@sk.ee, Common Name: EE Certification Centre Root CA, Organization: AS Sertifitseerimiskeskus, Country: EE
Package Control: Exported certificate: Common Name: Swisscom Root CA 1, Organizational Unit: Digital Certificate Services, Organization: Swisscom, Country: ch
Package Control: Exported certificate: Common Name: Certinomis - Autorité Racine, Organizational Unit: 0002 433998903, Organization: Certinomis, Country: FR
Package Control: Exported certificate: Common Name: GlobalSign, Organization: GlobalSign, Organizational Unit: GlobalSign ECC Root CA - R5
Package Control: Exported certificate: Common Name: QuoVadis Root CA 3 G3, Organization: QuoVadis Limited, Country: BM
Package Control: Exported certificate: Common Name: QuoVadis Root CA 1 G3, Organization: QuoVadis Limited, Country: BM
Package Control: Exported certificate: Common Name: Secure Global CA, Organization: SecureTrust Corporation, Country: US
Package Control: Exported certificate: Common Name: GlobalSign, Organization: GlobalSign, Organizational Unit: GlobalSign ECC Root CA - R4
Package Control: Exported certificate: Common Name: Symantec Class 3 Public Primary Certification Authority - G6, Organizational Unit: Symantec Trust Network, Organization: Symantec Corporation, Country: US
Package Control: Exported certificate: Common Name: UTN-USERFirst-Client Authentication and Email, Organizational Unit: http://www.usertrust.com, Organization: The USERTRUST Network, Locality: Salt Lake City, State/Province: UT, Country: US
Package Control: Exported certificate: Common Name: AddTrust Class 1 CA Root, Organizational Unit: AddTrust TTP Network, Organization: AddTrust AB, Country: SE
Package Control: Exported certificate: Common Name: IdenTrust Public Sector Root CA 1, Organization: IdenTrust, Country: US
Package Control: Exported certificate: Common Name: VeriSign Class 1 Public Primary Certification Authority - G3; Organizational Unit: (c) 1999 VeriSign, Inc. - For authorized use only, VeriSign Trust Network; Organization: VeriSign, Inc.; Country: US
Package Control: Exported certificate: Common Name: Entrust.net Certification Authority (2048); Organizational Unit: (c) 1999 Entrust.net Limited, www.entrust.net/CPS_2048 incorp. by ref. (limits liab.); Organization: Entrust.net
Package Control: Exported certificate: Common Name: DigiCert Global Root G2, Organizational Unit: www.digicert.com, Organization: DigiCert Inc, Country: US
Package Control: Exported certificate: Common Name: QuoVadis Root Certification Authority, Organizational Unit: Root Certification Authority, Organization: QuoVadis Limited, Country: BM
Package Control: Exported certificate: Common Name: Certum CA, Organization: Unizeto Sp. z o.o., Country: PL
Package Control: Exported certificate: Common Name: XRamp Global Certification Authority, Organization: XRamp Security Services Inc, Organizational Unit: www.xrampsecurity.com, Country: US
Package Control: Exported certificate: Common Name: Hellenic Academic and Research Institutions RootCA 2011, Organization: Hellenic Academic and Research Institutions Cert. Authority, Country: GR
Package Control: Exported certificate: Common Name: Cisco Root CA 2048, Organization: Cisco Systems
Package Control: Exported certificate: Common Name: Staat der Nederlanden Root CA - G3, Organization: Staat der Nederlanden, Country: NL
Package Control: Exported certificate: Common Name: GeoTrust Primary Certification Authority - G2, Organizational Unit: (c) 2007 GeoTrust Inc. - For authorized use only, Organization: GeoTrust Inc., Country: US
Package Control: Exported certificate: Common Name: OISTE WISeKey Global Root GA CA; Organizational Unit: OISTE Foundation Endorsed, Copyright (c) 2005; Organization: WISeKey; Country: CH
Exception in thread Thread-15:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "package_control.automatic_upgrader in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 111, in run
  File "package_control.automatic_upgrader in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 176, in install_missing
  File "package_control.package_manager in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 694, in list_available_packages
  File "package_control.package_manager in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 554, in _list_available
  File "package_control.package_manager in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 453, in list_repositories
  File "package_control.providers.channel_provider in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 201, in get_repositories
  File "package_control.providers.channel_provider in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 87, in fetch
  File "package_control.download_manager in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 340, in fetch
  File "package_control.downloaders.urllib_downloader in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 117, in download
  File "package_control.downloaders.urllib_downloader in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 317, in setup_opener
  File "package_control.ca_certs in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 31, in get_ca_bundle_path
  File "package_control.ca_certs in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 166, in get_system_ca_bundle_path
  File "package_control.deps.oscrypto.trust_list in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 122, in get_path
  File "package_control.deps.oscrypto._osx.trust_list in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 153, in extract_from_system
  File "package_control.deps.oscrypto._osx.trust_list in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 177, in _cert_callback
  File "package_control.ca_certs in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 108, in print_cert_subject
  File "package_control.deps.asn1crypto.x509 in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 1042, in human_friendly
  File "package_control.deps.asn1crypto.x509 in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 1071, in _recursive_humanize
  File "package_control.deps.asn1crypto.core in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 1100, in native
  File "package_control.deps.asn1crypto.core in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 1702, in native
  File "package_control.deps.asn1crypto.core in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 1672, in __unicode__
  File "package_control.deps.asn1crypto._teletex_codec in /Users/q/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 20, in decode
UnicodeDecodeError: 'charmap' codec can't decode byte 0x23 in position 32: character maps to <undefined>
```

**Comment by collinzh** [comment]
I think this issue is being discussed in #1267

**Comment by mgrandi** [comment]
correct, it is probably caused by an openVPN cert that got installed, that has something along the lines of `'http://openvpn.net/localca.html #1433234179'` for the subject name, deleting those keys in keychain access will make it work again

**Comment by wbond** [comment]
Please try following the instructions at https://github.com/wbond/package_control/issues/1293#issuecomment-342595937 and let me know on that issue if you have any trouble.

**Comment by wbond** [comment]
This seems to be solved by 3.3.1-beta1.
